### PR TITLE
[CWS] Add functional tests for activity dumps

### PR DESCRIPTION
**Pull request by spikat**
### What does this PR do?

This add first functional tests for activity dumps.

This validates only the activity dumps generated manually based on a given comm.

There is two checks, one to validate the DNS request presence, and another one to
validate the presence of bound sockets.

For each of them, we validate the JSON output through a ~generated schema, and the content of the MSGP output.

A lot of work is still to be done:
- Validate AD generation based on cgroups
- Validate the AD content regarding processes (exec) and opened files
- Validate the content of snapshot (not at runtime)
- Validate the generated profiles and graphs
- Validate the core things (compression, timeout, sending, ...)

### Motivation

Make the firsts little step to be able to test activity dumps.

### Describe how to test/QA your changes

inv -e security-agent.functional-tests --testflags "-test.v -test.run TestActivityDumps"

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by vickenty** [comment]
Does not seem to have anything for agent-core to review.

**Review by jtappa — APPROVED** [review]
👍🏻 from docs team

**Review by lebauce — APPROVED** [review]

File: pkg/security/tests/activity_dumps_test.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	"os/exec"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/security/secl/rules"
+)
+
+func TestActivityDumps(t *testing.T) {
+	ruleDefs := []*rules.RuleDefinition{
```
Comment (lebauce) [inline pkg/security/tests/activity_dumps_test.go:24]: Should not be necessary

File: pkg/security/tests/activity_dumps_test.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+			case ".json":
+				if jsonOK == true {
+					t.Fatal("Got more than one JSON file:", outputFiles)
+				}
+				content, err := os.ReadFile(f)
+				if err != nil {
+					t.Fatal(err)
+				}
+				if !validateActivityDumpSchema(t, string(content)) {
+					t.Fatal(string(content))
```
Comment (lebauce) [inline pkg/security/tests/activity_dumps_test.go:82]: nit: t.Error

Comment (spikat) [reply pkg/security/tests/activity_dumps_test.go]: fixed, thanks

File: pkg/security/tests/activity_dumps_test.go (hunk 1 above)
Comment (spikat) [reply pkg/security/tests/activity_dumps_test.go:24]: will be fixed by https://github.com/DataDog/datadog-agent/pull/12623,  I will rebase/remove this fake rule when the PR will be merged

**Pull request merged**